eva_cttv_pipeline/trait_mapping/trait_names_parsing.py

```diff
@@ -2,67 +2,46 @@
-    # ClinVar data. Originally "clinvarSet" was the top level, but this level was removed and
-    # referenceClinVarAssertion is now the top level.
-    if "clinvarSet" in clinvar_json:
-        trait_set = clinvar_json["clinvarSet"]["referenceClinVarAssertion"]["traitSet"]
-    else:
-        trait_set = clinvar_json["referenceClinVarAssertion"]["traitSet"]
-    trait_names_to_return = []
-    for trait in trait_set['trait']:
-        trait_name = None
-        for name in trait['name']:
-            # First trait name in the list will always be the "Preferred" one
-            if name['elementValue']['type'] == 'Preferred':
-                trait_name = name['elementValue']['value']
-                break
-            elif name['elementValue']['type'] in ["EFO URL", "EFO id", "EFO name"]:
-                continue  # if the trait name not originally from clinvar
-            elif trait_name is None:
-                trait_name = name['elementValue']['value']
-        if trait_name is not None:
-            trait_names_to_return.append(trait_name.lower())
-
-    return trait_names_to_return
-
-
 def parse_trait_names(filepath: str) -> list:
     """
-    For a file containing ClinVar records in the format of one json per line, return a list of the
-    trait names for the records in the file.
+    For a file containing ClinVar records in the TSV format, return a list of the trait names for the records in the
+    file.
 
-    :param filepath: String giving the path to a gzipped file containing jsons of ClinVar records,
-                     one per line.
-    :return: List (important that it is a list and not a set, it is used to calculate the
-             frequencies) containing the trait names of the traits of the records in the file
-             containing ClinVar records.
+    :param filepath: Path to a gzipped file containing ClinVar TSV summary.
+    :return: A list of traits. It is important that it is a list and not a set, as it is used to calculate the
+        frequencies of each trait later on.
     """
+
+    # This is a dict of sets we use to keep track on which alleles are linked to which traits. The reason this is done
+    # is because some (most) alleles appear twice in the document with coordinates for GRCh37 and GRCh38, and we don't
+    # want to count them twice.
+    trait_names_by_allele_id = dict()
+
+    with gzip.open(filepath, "rt") as clinvar_summary:
+        header = clinvar_summary.readline().rstrip().split('\t')
+        for line in clinvar_summary:
+            values = line.rstrip().split('\t')
+            data = dict(zip(header, values))
+
+            # Check if the record should be processed given its level of clinical significance
+            acceptable_clinical_significance_present = False
+            for clinical_significance in data['ClinicalSignificance'].split(','):
+                # TODO: unify the acceptable clinical significance levels project-wide
```

Comment:
I am ok with the TODO but could you locally extract the list to a string and place it as a constant in this file for easy reference?

Reply:
Yes, I agree. Fixed in https://github.com/EBIvariation/eva-cttv-pipeline/pull/99/commits/78792b58e55f0334562892cab392632a82c964f7